pkg/scaling/resolver/scale_resolvers.go

```diff
@@ -286,7 +287,7 @@ func getTriggerAuthSpec(ctx context.Context, client client.Client, triggerAuthRe
 
 func resolveEnv(ctx context.Context, client client.Client, logger logr.Logger, container *corev1.Container, namespace string, secretsLister corev1listers.SecretLister) (map[string]string, error) {
 	resolved := make(map[string]string)
-
+	secretAccessRestricted, _ := strconv.ParseBool(util.GetRestrictSecretAccess())
```

Comment:
Thanks @Indresh2410 
I think it's safer to use `isSecretAccessRestricted(logger)` instead of parsing the prop here.